README.md

```diff
@@ -162,4 +177,142 @@ $ oc get image | grep onload
+default-route-openshift-image-registry.apps.test.kube.test/onload-clusterlocal/onload-device-plugin  latest
+default-route-openshift-image-registry.apps.test.kube.test/onload-clusterlocal/onload-user           git27b3826
+default-route-openshift-image-registry.apps.test.kube.test/openshift-kmm/sfc-module                  git27b3826-4.18.0-372.49.1.el8_6.x86_64
+```
+
+### Getting the images
+
+The images can either be built using podman directly (untested) or using a
+non-airgapped cluster.
+
+
+#### Pulling images from a cluster
+First follow the section about [logging in via podman](#podman-login) below.
+
+The images can be pulled via the following command:
+```
+$ sudo podman pull ${REGHOST}/openshift-kmm/sfc-module:git27b3826-4.18.0-372.49.1.el8_6.x86_64 --tls-verify=false
+```
+
+Repeat this process for each image you want to pull from the cluster.
+The image specification to pull from the cluster should be of the form:
+```
+REGISTRY/OPENSHIFT_NAMESPACE/IMAGE_NAME:IMAGE_TAG
+```
+
+`podman save` can be used to save the into a local file, e.g:
+```
+$ sudo podman save -o images/sfc.tar default-route-openshift-image-registry.apps.test.kube.test/openshift-kmm/sfc-module:git27b3826-4.18.0-372.49.1.el8_6.x86_64
+```
+which will write the image into `images/sfc.tar`. Then use `podman load` to
+load the image from the written file:
+```
+$ sudo podman load -i images.tar
+```
+
+### Podman login
+
+In order to be able to pull/push images into the cluster's image registry you
+must log in with podman.
+
+1. log in to openshift cluster
+```
+$ oc login -u kubeadmin -p PASSWORD
+```
+2. Get the name of the image registry
+```
+oc get route default-route -n openshift-image-registry --template='{{ .spec.host }}'
+```
+This can be stored in a local variable
+```
+REGHOST=`oc get route default-route -n openshift-image-registry --template='{{ .spec.host }}'`
+```
+3. Actually login to the image registry with podman
+```
+sudo podman login -u kubeadmin -p $(oc whoami -t) ${REGHOST} --tls-verify=false
+```
+
+#### A Note on `--tls-verify=false`
+
+Without doing any extra steps podman will complain about the certificates used
+by the image registry. For development purposes `--tls-verify=false` can be
+used to bypass this issue. A proper solution would be to extract the
+certificates and pass these to podman.
+
+### Set up the imagestreams and namespaces
+
+```
+$ oc create -f sfc/imagestream/imagestream.yaml
+$ oc create -f onload/imagestream/
+
+$ oc create -f sfptpd/sfptpd-imagestream.yaml
+
+$ oc create -f examples/sfnettest/0000-imagestream.yaml
+```
+
+### Pushing the images to the cluster's image-registry
+
+For each image you want to push run the command:
+```
+$sudo podman push REGISTRY/OPENSHIFT_NAMESPACE/IMAGE_NAME:TAG --tls-verify=false
+```
+
+### Deploying sfc driver
+
+#### KMM deployment
+First remove the in-tree driver from the worker nodes
+```
+# rmmod sfc
+```
+This should allow the kmm to install the kernel module without issue.
```

Comment:
If this command is required it should be added to `sfc/mco/99-sfc-machineconfig.bu` rather than left as an exercise for the user.

Reply:
Removing `sfc` and `sfc_driverlink` is already handled in [`sfc-replace-kernel-module.sh#L9`](https://github.com/Xilinx-CNS/kubernetes-onload/blob/e556d4d5186cca334ccf56424263f658bd6e6036/sfc/mco/sfc-replace-kernel-module.sh#L9).

This is only necessary since we are using KMM v1.0 (which is the only one I can find on the operator hub). If we were using KMM v1.1 there is the `inTreeModuleToRemove` field which would handle the removal for us.

Reply:
Good point, this is useful information to have in that section of the documentation. 

~~~suggestion
#### KMM Module deployment

If KMM v1.0 is used, the in-tree driver must be removed from the worker nodes
before deployment. This is a one-time action which is automated by the
MachineConfig component upon reboot. In KMM v1.1 onwards, add 
`inTreeModuleToRemove: sfc` to the `sfc` Module CR.

```sh
rmmod sfc
```
~~~

Reply:
Your suggestions looks good, I've added it as a separate commit